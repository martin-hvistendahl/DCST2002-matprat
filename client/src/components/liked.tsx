import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Cards, Rows, RecipeView } from '.././widgets';
import { NavLink } from 'react-router-dom';
import service, { Recipe } from '.././service';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student
export class LikedRecipes extends Component {
  recipes: Recipe[] = [];

  render() {
    return (
      <div className="margintop">
        <div className="container-fluid">
          <Rows>{/* Filtrerer recipes etter hvilke som er likt og viser de med et card som linker til oppskriften */}
            {this.recipes
              .filter((recipe) => recipe.liked == true)
              .map((likedRecipe) => (
                <Cards title="" key={likedRecipe.oppskrift_id}>
                  <NavLink className="black" to={'/recipe/' + likedRecipe.oppskrift_id}>
                    <RecipeView
                      img={likedRecipe.bilde_adr}
                      name={likedRecipe.oppskrift_navn}
                      numbOfPors={likedRecipe.ant_pors}
                    ></RecipeView>
                  </NavLink>
                </Cards>
              ))}
          </Rows>
        </div>
      </div>
    );
  }

  mounted() {
    /* Henter alle oppskrftene */
    service
      .getAllRepice()
      .then((recipes) => {
        this.recipes = recipes;
      })
      .catch((error) => Alert.danger('Error getting tasks: ' + error.message));
  }
}
