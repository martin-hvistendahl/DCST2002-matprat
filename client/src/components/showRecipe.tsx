import * as React from 'react';
import { Component } from 'react-simplified';
import { Alert, Card, Row, Button, Cards, Rows, RecipeView } from '.././widgets';
import { NavLink } from 'react-router-dom';
import service, { Category, Ingredient, Recipe, Recipe_Content } from '.././service';
import { createHashHistory } from 'history';

const history = createHashHistory(); // Use history.push(...) to programmatically change path, for instance after successfully saving a student
export class ShowRecipe extends Component<{ match: { params: { id: number } } }> {
  recipe: Recipe = {
    oppskrift_id: 0,
    oppskrift_navn: '',
    oppskrift_beskrivelse: '',
    oppskrift_steg: '',
    ant_pors: 0,
    bilde_adr: '',
    kategori_id: 0,
    land_id: 0,
    ant_like: 0,
    liked: false,
  };
  portions: number = 0;
  recipeContent: Recipe_Content[] = [];
  ingredients: Ingredient[] = [];
  categories: Category[] = [];
  allRecipes: Recipe[] = [];
  recommendedRecipes: Recipe[] = [];

  render() {
    return (
      <div className="margintop">
        <Card title="">
          <img className="stort" src={this.recipe.bilde_adr}></img>
          <h1>{this.recipe.oppskrift_navn}</h1>
          <br />
          {this.recipe.oppskrift_beskrivelse != '' ? (
            <p>Beskrivelse: {this.recipe.oppskrift_beskrivelse}</p>
          ) : (
            ''
          )}
          <p className="font-weight-bold" style={{ marginBottom: '0px' }}>
            Kategori:{' '}
            {
              this.categories.find((kategori) => kategori.kategori_id == this.recipe.kategori_id)
                ?.kategori_navn
            }
          </p>
         
          <input
            type="checkbox"
            id="checkbox"
            checked={this.recipe.liked}
            onChange={() => {
              service.likeRecipe(this.recipe.oppskrift_id, !this.recipe.liked).then(() => {
                ShowRecipe.instance()?.mounted();
              });
            }}
          />
          <label htmlFor="checkbox">
            <svg id="heart-svg" viewBox="467 392 58 57" xmlns="http://www.w3.org/2000/svg">
              <g id="Group" fill="none" fillRule="evenodd" transform="translate(467 392)">
                <path
                  d="M29.144 20.773c-.063-.13-4.227-8.67-11.44-2.59C7.63 28.795 28.94 43.256 29.143 43.394c.204-.138 21.513-14.6 11.44-25.213-7.214-6.08-11.377 2.46-11.44 2.59z"
                  id="heart"
                  fill="#AAB8C2"
                />
                <circle id="main-circ" fill="#E2264D" opacity="0" cx="29.5" cy="29.5" r="1.5" />

                <g id="grp7" opacity="0" transform="translate(7 6)">
                  <circle id="oval1" fill="#9CD8C3" cx="2" cy="6" r="2" />
                  <circle id="oval2" fill="#8CE8C3" cx="5" cy="2" r="2" />
                </g>

                <g id="grp6" opacity="0" transform="translate(0 28)">
                  <circle id="oval1" fill="#CC8EF5" cx="2" cy="7" r="2" />
                  <circle id="oval2" fill="#91D2FA" cx="3" cy="2" r="2" />
                </g>

                <g id="grp3" opacity="0" transform="translate(52 28)">
                  <circle id="oval2" fill="#9CD8C3" cx="2" cy="7" r="2" />
                  <circle id="oval1" fill="#8CE8C3" cx="4" cy="2" r="2" />
                </g>

                <g id="grp2" opacity="0" transform="translate(44 6)">
                  <circle id="oval2" fill="#CC8EF5" cx="5" cy="6" r="2" />
                  <circle id="oval1" fill="#CC8EF5" cx="2" cy="2" r="2" />
                </g>

                <g id="grp5" opacity="0" transform="translate(14 50)">
                  <circle id="oval1" fill="#91D2FA" cx="6" cy="5" r="2" />
                  <circle id="oval2" fill="#91D2FA" cx="2" cy="2" r="2" />
                </g>

                <g id="grp4" opacity="0" transform="translate(35 50)">
                  <circle id="oval1" fill="#F48EA7" cx="6" cy="5" r="2" />
                  <circle id="oval2" fill="#F48EA7" cx="2" cy="2" r="2" />
                </g>

                <g id="grp1" opacity="0" transform="translate(24)">
                  <circle id="oval1" fill="#9FC7FA" cx="2.5" cy="3" r="2" />
                  <circle id="oval2" fill="#9FC7FA" cx="7.5" cy="2" r="2" />
                </g>
              </g>
            </svg>
          </label>
          <p>Antall likes: {this.recipe.ant_like}</p>
          <h3>Oppskrift:</h3>
          <br />
          <pre>{this.recipe.oppskrift_steg}</pre>
          <h3>Ingredienser:</h3>
          <br />
          {this.recipeContent.map((rc, i) => (
            <Row key={i}>
              <p style={{ width: '250px' }}>
                {i + 1}. {/* find name of ingridient */}
                {this.ingredients.find((ing) => ing.ingred_id == rc.ingred_id)?.ingred_navn}
                {/* {this.ingredients.filter((ing) => rc.ingred_id == ing.ingred_id)[0].ingred_navn}{' '} */}
              </p>
              <p style={{ width: '75px' }}>
                {((Number(rc.mengde) * this.portions) / this.recipe.ant_pors).toFixed(1)}
              </p>
              <p style={{ width: '130px' }}>{rc.maleenhet}</p>
            </Row>
          ))}
          Porsjoner:{' '}
          <Button.Danger id="btnDec" onClick={this.decrementPortions}>
            -
          </Button.Danger>{' '}
          <b>{this.portions}</b>{' '}
          <Button.Success id="btnInc" onClick={this.incrementPortions}>
            +
          </Button.Success>
        </Card>
        <a href={`mailto:?subject=${this.recipe.oppskrift_navn}&body=${this.recipe.oppskrift_beskrivelse} %0d%0a ${window.location.href}`}>Heidu</a>
        <Button.Success onClick={() => history.push('/recipe/edit/' + this.props.match.params.id)}>
          Endre oppskrift
        </Button.Success>
        <Button.Danger
          //@ts-ignore
          id="deleteRecipe"
          onClick={() => this.deleteRecipe(this.props.match.params.id)}
        >
          Slett oppskrift
        </Button.Danger>
        <Button.Success id="btnSend" onClick={this.ingredientsToShoppingList}>
          Send ingredienser til handleliste
        </Button.Success>
        {/* Recomend 5 recipes based on the category */}
        <h3>Andre oppskrifter i samme kategori:</h3>
        <Rows>
          {this.recommendedRecipes.map((recipe) => (
            <Cards title="" key={recipe.oppskrift_id}>
              <NavLink className="black" to={'/recipe/' + recipe.oppskrift_id}>
                <RecipeView
                  img={recipe.bilde_adr}
                  name={recipe.oppskrift_navn}
                  numbOfPors={recipe.ant_pors}
                ></RecipeView>
              </NavLink>
            </Cards>
          ))}
        </Rows>
      </div>
    );
  }

  mounted() {
    service
      .getRecipe(this.props.match.params.id)
      .then((recipe) => {
        this.recipe = recipe[0];
        this.portions = recipe[0].ant_pors;
      })
      .catch((error) => Alert.danger('Error getting recipe: ' + error.message));

    service
      .getAllIngredient()
      .then((ingredients) => (this.ingredients = ingredients))
      .catch((error) => Alert.danger('Error getting ingredients: ' + error.message));

    service
      .getRecipeContent(this.props.match.params.id)
      .then((recipeContent) => (this.recipeContent = recipeContent))
      .catch((error) => Alert.danger('Error getting recipe content: ' + error.message));

    service
      .getAllCategory()
      .then((categories) => (this.categories = categories))
      .catch((error) => Alert.danger('Error getting categories: ' + error.message));

    service
      .getAllRepice()
      .then((recipes) => {
        this.allRecipes = recipes;
        this.findRecommendedRecipes(recipes.length);
      })
      .catch((error) => Alert.danger('Error getting recipes: ' + error.message));
  }

  findRecommendedRecipes(lengtCheck: number) {
    if (this.allRecipes.length == lengtCheck && this.recipe.kategori_id != 0) {
      this.recommendedRecipes = this.allRecipes
        .filter((recipe) => recipe.kategori_id == this.recipe.kategori_id)
        .slice(0, 5);
    } else {
      this.findRecommendedRecipes(lengtCheck);
    }
  }
  incrementPortions() {
    this.portions++;
  }
  decrementPortions() {
    if (this.portions > 1) {
      this.portions--;
    }
  }
  deleteRecipe(id: number) {
    service
      .deleteRecipe(id)
      .then(() => history.push('/'))
      .catch((error) => Alert.danger('Error deleting recipe: ' + error.message));
  }
  ingredientsToShoppingList() {
    this.recipeContent.forEach((rc) => {
      const ingredient = {
        ingred_id: rc.ingred_id,
        mengde: (Number(rc.mengde) * this.portions) / this.recipe.ant_pors,
        maleenhet: rc.maleenhet,
      };
      service.addIngredient(ingredient)
      .then(() => {history.push('/shoppinglist');})
    });
    
  }
}
