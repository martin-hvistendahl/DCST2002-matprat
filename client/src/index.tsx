import ReactDOM from 'react-dom';
import * as React from 'react';
import { Component } from 'react-simplified';
import { NavLink, HashRouter, Route } from 'react-router-dom';
import {
  NavBar,
  Car,
  Card,
  Cards,
  Alert,
  Column,
  Row,
  Form,
  Button,
  RecipeView,
  Oppskrifter,
  Mat,
} from './widgets';
import { NewRecipe } from './components/newRecipe';
import { EditRecipe } from './components/editRecipe';
import { ShowRecipe } from './components/showRecipe';
import { LikedRecipes } from './components/liked';
import { ShoppingList } from './components/shoppingList';
import { Icebox } from './components/icebox';
import { ShowAllRecipe } from './components/showAllRecipe';

import service, { Recipe } from './service';
import { createHashHistory } from 'history';

export class Menu extends Component {
  render() {
    return (
      <>
        <div className="header">
          <NavBar brand="Hjem">
            <NavBar.Link to="/showallrecipe">Alle oppskrifter</NavBar.Link>
            <NavBar.Link to="/newrecipe">Ny oppskrift</NavBar.Link>
            <NavBar.Link to="/liked">Liked</NavBar.Link>
            <NavBar.Link to="/shoppinglist">Handleliste</NavBar.Link>
            <NavBar.Link to="/icebox">Kjøleskap</NavBar.Link>
          </NavBar>
        </div>
      </>
    );
  }
}
export class Home extends Component {
  isMounted = false;
  suggestedRecipe: Recipe[] = [];
  recipes: Recipe[] = [];
  suggestedRecipeList: Recipe[] = [];

  likedFromCountrey: [] = [];
  likedFromCategory: [] = [];

  //random number under here from 0 to recipes.length
  render() {
    //finds the recipes that are liked and adds the country and category id to the arrays
    let random: number = Math.floor(Math.random() * this.recipes.length);

    return (
      <>
        {this.isMounted ? (
          <Card title="">
            <div className="frontpage">
              <h1>Anbefalt oppskrift:</h1>
              <br></br>
              <div className={'recipeToDay'}>
                {this.recipes.length != 0
                  ? this.recipes
                      .filter((recipes, i) => i == random)
                      .map((recipe, rei) => (
                        <div key={rei}>
                          <NavLink className="black" to={'/recipe/' + recipe.oppskrift_id}>
                            <img src={recipe.bilde_adr} className="frontPicture" alt="recipe" />
                            <br />
                            <br />
                            <h3 id="frontname" style={{ color: 'black' }}>
                              {recipe.oppskrift_navn}
                            </h3>
                          </NavLink>
                        </div>
                      ))
                  : ''}
              </div>
              <br />
              <br />
              <div>
                <div title="Anbefalte oppskrifter basert på dine likte:">
                  Anbefalte oppskrifter basert på det du liker
                  <center>
                    <Row>
                      {this.recipes.length != 0
                        ? this.suggestedRecipeList.map((likedRecipe) => (
                            <Cards>
                              <NavLink className="black" to={'/recipe/' + likedRecipe.oppskrift_id}>
                                <RecipeView
                                  img={likedRecipe.bilde_adr}
                                  name={likedRecipe.oppskrift_navn}
                                  numbOfPors={likedRecipe.ant_pors}
                                ></RecipeView>
                              </NavLink>
                            </Cards>
                          ))
                        : ''}
                    </Row>
                  </center>
                  <NavBar.Link to={'/showallrecipe'} style={{ width: '130px' }}>
                    Alle oppskrifter
                  </NavBar.Link>
                </div>
              </div>
            </div>
          </Card>
        ) : (
          ''
        )}
      </>
    );
  }
  mounted() {
    this.isMounted = false;
    service
      .getAllRepice()
      .then((recipes) => {
        this.recipes = recipes;
        if (this.recipes.filter((recipe) => recipe.liked == true).length <= 0) {
          //loop five times to get five random recipes
          for (let i = 0; i < 5; i++) {
            this.suggestedRecipeList.push(
              this.recipes[Math.floor(Math.random() * this.recipes.length)]
            );
          }
        } else {
          this.recipes
            .filter((recipe) => recipe.liked == true)
            .map(
              (likedRecipe) => (
                //@ts-ignore
                this.likedFromCountrey.push(likedRecipe.land_id),
                //@ts-ignore
                this.likedFromCategory.push(likedRecipe.kategori_id)
              )
            );
          console.log(this.likedFromCountrey, this.likedFromCategory);

          this.recipes.map((element) => {
            //@ts-ignore
            console.log(this.likedFromCountrey.includes(element.land_id));
            if (
              //@ts-ignore
              this.likedFromCountrey.includes(element.land_id) &&
              //@ts-ignore
              this.likedFromCategory.includes(element.kategori_id) &&
              element.liked == false
            ) {
              this.suggestedRecipe.push(element);
            }
          });

          for (let i = 0; i < this.recipes.length; i++) {
            //@ts-ignore
            if (this.likedFromCategory.includes(this.recipes[i].kategori_id)) {
              this.suggestedRecipe.push(this.recipes[i]);
            } else {
            }
          }

          for (let i = 0; i < 5; i++) {
            //random number from suggestedRecipe
            let random = Math.floor(Math.random() * this.suggestedRecipe.length);
            this.suggestedRecipeList.push(this.suggestedRecipe[random]);

            this.suggestedRecipe.splice(random, 1);
          }
          console.log(this.suggestedRecipeList);
        }
      })
      .catch((error) => Alert.danger('Error getting tasks: ' + error.message));
    this.isMounted = true;
  }
}

ReactDOM.render(
  <HashRouter>
    <div>
      <Alert />
      <Menu />
      <Route exact path="/" component={Home} />
      <Route exact path="/showallrecipe" component={ShowAllRecipe} />
      <Route exact path="/newrecipe" component={NewRecipe} />
      <Route exact path="/recipe/:id" component={ShowRecipe} />
      <Route exact path="/recipe/edit/:id" component={EditRecipe} />
      <Route exact path="/shoppinglist" component={ShoppingList} />
      <Route exact path="/liked" component={LikedRecipes}></Route>
      <Route exact path="/icebox" component={Icebox}></Route>
    </div>
  </HashRouter>,
  document.getElementById('root') || document.createElement('div')
);
