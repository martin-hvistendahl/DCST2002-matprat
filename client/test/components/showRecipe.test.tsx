import * as React from 'react';

import { shallow } from 'enzyme';
import { Alert, Button } from '../../src/widgets';
import { ShowRecipe } from '../../src/components/showRecipe';

import { createHashHistory } from 'history';

jest.mock('../../src/service', () => {
  class Service {
    getAllCountry() {
      return Promise.resolve([
        {
          land_id: 1,
          land_navn: 'Sverige',
        },
        {
          land_id: 2,
          land_navn: 'Italia',
        },
      ]);
    }
    getAllCategory() {
      return Promise.resolve([
        {
          kategori_id: 1,
          kategori_navn: 'Ikea mat',
        },
        {
          kategori_id: 2,
          kategori_navn: 'enkelt',
        },
      ]);
    }
    getAllIngredient() {
      return Promise.resolve([
        {
          ingred_id: 1,
          ingred_navn: 'pizzadeig',
        },
        {
          ingred_id: 2,
          ingred_navn: 'pizza fyll',
        },
        {
          ingred_id: 3,
          ingred_navn: 'kjøttboller',
        },
      ]);
    }
    getAllRepice(id: number) {
      return Promise.resolve([
        {
          oppskrift_id: 1,
          oppskrift_navn: 'Pizza',
          oppskrift_beskrivelse: 'Pizza er god og enkel',
          oppskrift_steg: 'Bland deigen og la den heve',
          ant_pors: 4,
          bilde_adr: 'pizza.jpg',
          kategori_id: 2,
          land_id: 2,
          ant_like: 1,
          liked: true,
        },
        {
          oppskrift_id: 2,
          oppskrift_navn: 'Kjøttboller i brunsaus',
          oppskrift_beskrivelse: 'Kjøttboller er god og enkel',
          oppskrift_steg: 'dra til ikea og kjøp kjøttboller',
          ant_pors: 4,
          bilde_adr: 'kjøttboller.jpg',
          kategori_id: 1,
          land_id: 3,
          ant_like: 0,
          liked: false,
        },
      ]);
    }
    getRecipe(id: number) {
      return Promise.resolve([
        {
          oppskrift_id: 1,
          oppskrift_navn: 'Pizza',
          oppskrift_beskrivelse: 'Pizza er god og enkel',
          oppskrift_steg: 'Bland deigen og la den heve',
          ant_pors: 4,
          bilde_adr: 'pizza.jpg',
          kategori_id: 2,
          land_id: 2,
          ant_like: 1,
          liked: true,
        },
      ]);
    }

    getRecipeContent(id: number) {
      return Promise.resolve([
        {
          oppskrift_id: 1,
          ingred_id: 1,
          mengde: 1,
          maleenhet: 'stk',
        },
        {
          oppskrift_id: 1,
          ingred_id: 2,
          mengde: 1,
          maleenhet: 'håndfull',
        },
        {
          oppskrift_id: 1,
          ingred_id: 3,
          mengde: 400,
          maleenhet: 'g',
        },
      ]);
    }
    deleteRecipe(id: number) {
      return Promise.resolve();
    }
    createIngredient() {
      return Promise.resolve();
    }
    addIngredient() {
      return Promise.resolve();
    }
    createRecipe() {
      return Promise.resolve(1);
    }
    likeRecipe(id: number, like: boolean) {
      return Promise.resolve();
    }
  }
  return new Service();
});

describe('ShowRecipe tests', () => {
  test('Send ingredient to shopping list', (done) => {
    const history = createHashHistory();
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);

    wrapper.find('#btnSend').simulate('click');

    setTimeout(() => {
      //ved å sjekke om history length er 2 så vet vi at vi har navigert til en annen side enn den orginale
      // expect(history.location.pathname).toBe('/shoppinglist');
      expect(window.location.href).not.toEqual('http://localhost/#/recipe/1');

      done();
    });
  });
  test('Delete recipe button works', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 2 } }} />);
    setTimeout(() => {
      wrapper.find('#deleteRecipe').simulate('click');
      setTimeout(() => {
        expect(window.location.href).toEqual('http://localhost/#/');
        done();
      });
    });
  });
  //test if downlaad button works, this test gives an error because it is not possible to test the download function and more but i tried my best
  test('Download button works', (done) => {
    //@ts-ignore
    window.URL.createObjectURL = function () {};
    window.URL.revokeObjectURL = function () {};
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    const wrapperAlert = shallow(<Alert />);
    setTimeout(() => {
      wrapper.find('#downloadPageBtn').simulate('click');
      setTimeout(() => {
        expect(
          wrapperAlert.containsMatchingElement(
            <div>
              Oppskriten ble lastet ned<button></button>
            </div>
          )
        ).toEqual(true);
        done();
      });
    });
  });
  test('Edit recipe button works', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      wrapper.find('#btnEdit').simulate('click');
      expect(window.location.href).toEqual('http://localhost/#/recipe/edit/1');
      done();
    });
  });

  test('Like recipe button works', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      wrapper.find('#checkbox').simulate('change', { target: { checked: false } });
      //kan ikke expect false fordi vi gjør et database call og det klarer ikke jest
      expect(wrapper.find('#checkbox').props().checked).toEqual(true);

      done();
    });
  });
  test('Draws correctly', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      expect(wrapper).toMatchSnapshot();
      expect(
        wrapper.containsMatchingElement([<h1>Pizza</h1>, <p>Beskrivelse:Pizza er god og enkel</p>])
      ).toBe(true);
      done();
    });
  });

  test('Number increment works', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      wrapper.find('#btnInc').simulate('click');

      expect(wrapper).toMatchSnapshot();
      expect(
        wrapper.containsMatchingElement([
          <b>5</b>,
          <p> 1. pizzadeig 1.3 stk</p>,
          <p> 2. pizza fyll 1.3 håndfull</p>,
          <p> 3. kjøttboller 500.0 g</p>,
        ])
      ).toBe(true);

      done();
    });
  });

  test('Number decrement works', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);
    setTimeout(() => {
      wrapper.find('#btnDec').simulate('click');

      expect(wrapper).toMatchSnapshot();
      expect(
        wrapper.containsMatchingElement([
          <b>3</b>,
          <p> 1. pizzadeig 0.75 stk</p>,
          <p> 2. pizza fyll 0.75 håndfull</p>,
          <p> 3. kjøttboller 300.0 g</p>,
        ])
      ).toBe(true);
      done();
    });
  });

  test('Number of likes', (done) => {
    const wrapper = shallow(<ShowRecipe match={{ params: { id: 1 } }} />);

    setTimeout(() => {
      wrapper.find('#checkbox').simulate('change', { target: { checked: true } });

      expect(wrapper).toMatchSnapshot();
      expect(wrapper.containsMatchingElement([<p>Antall likes: 2</p>])).toBe(true);
      done();
    });
  });
});
