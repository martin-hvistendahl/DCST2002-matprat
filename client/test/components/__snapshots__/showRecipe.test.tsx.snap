// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowRecipe tests Draws correctly 1`] = `
<div>
  <Card
    title=""
  >
    <img
      src="pizza.jpg"
    />
    <h1>
      Pizza
    </h1>
    <p>
      Beskrivelse: 
      Pizza er god og enkel
    </p>
    <p>
      Kategori:
       
      enkelt
    </p>
    <p>
      Antall likes: 
      1
    </p>
    <FormCheckbox
      checked={false}
      id="checkbox"
      onChange={[Function]}
    />
    <label
      htmlFor="checkbox"
      id="heart"
    >
      <img
        src="https://static5.depositphotos.com/1001759/463/i/600/depositphotos_4634746-stock-photo-red-heart-isolated-white-background.jpg"
        width="50px"
      />
    </label>
    <h5>
      Oppskrift:
    </h5>
    <pre>
      Bland deigen og la den heve
    </pre>
    <h3>
      Ingredienser
    </h3>
    Porsjoner
     
    <ButtonDanger
      id="btnDec"
      onClick={[Function]}
    >
      -
    </ButtonDanger>
     
    <b>
      4
    </b>
     
    <ButtonSuccess
      id="btnInc"
      onClick={[Function]}
    >
      +
    </ButtonSuccess>
    <p
      key="0"
    >
      1
      .
       
      pizzadeig
       
      1.0
       
      stk
    </p>
    <p
      key="1"
    >
      2
      .
       
      pizza fyll
       
      1.0
       
      håndfull
    </p>
    <p
      key="2"
    >
      3
      .
       
      kjøttboller
       
      400.0
       
      g
    </p>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Endre oppskrift
  </ButtonSuccess>
  <ButtonDanger
    id="deleteRecipe"
    onClick={[Function]}
  >
    Slett oppskrift
  </ButtonDanger>
  <ButtonSuccess
    id="btnSend"
    onClick={[Function]}
  >
    Send ingredienser til handleliste
  </ButtonSuccess>
</div>
`;

exports[`ShowRecipe tests Number decrement works 1`] = `
<div>
  <Card
    title=""
  >
    <img
      src="pizza.jpg"
    />
    <h1>
      Pizza
    </h1>
    <p>
      Beskrivelse: 
      Pizza er god og enkel
    </p>
    <p>
      Kategori:
       
      enkelt
    </p>
    <p>
      Antall likes: 
      1
    </p>
    <FormCheckbox
      checked={false}
      id="checkbox"
      onChange={[Function]}
    />
    <label
      htmlFor="checkbox"
      id="heart"
    >
      <img
        src="https://static5.depositphotos.com/1001759/463/i/600/depositphotos_4634746-stock-photo-red-heart-isolated-white-background.jpg"
        width="50px"
      />
    </label>
    <h5>
      Oppskrift:
    </h5>
    <pre>
      Bland deigen og la den heve
    </pre>
    <h3>
      Ingredienser
    </h3>
    Porsjoner
     
    <ButtonDanger
      id="btnDec"
      onClick={[Function]}
    >
      -
    </ButtonDanger>
     
    <b>
      3
    </b>
     
    <ButtonSuccess
      id="btnInc"
      onClick={[Function]}
    >
      +
    </ButtonSuccess>
    <p
      key="0"
    >
      1
      .
       
      pizzadeig
       
      0.8
       
      stk
    </p>
    <p
      key="1"
    >
      2
      .
       
      pizza fyll
       
      0.8
       
      håndfull
    </p>
    <p
      key="2"
    >
      3
      .
       
      kjøttboller
       
      300.0
       
      g
    </p>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Endre oppskrift
  </ButtonSuccess>
  <ButtonDanger
    id="deleteRecipe"
    onClick={[Function]}
  >
    Slett oppskrift
  </ButtonDanger>
  <ButtonSuccess
    id="btnSend"
    onClick={[Function]}
  >
    Send ingredienser til handleliste
  </ButtonSuccess>
</div>
`;

exports[`ShowRecipe tests Number increment works 1`] = `
<div>
  <Card
    title=""
  >
    <img
      src="pizza.jpg"
    />
    <h1>
      Pizza
    </h1>
    <p>
      Beskrivelse: 
      Pizza er god og enkel
    </p>
    <p>
      Kategori:
       
      enkelt
    </p>
    <p>
      Antall likes: 
      1
    </p>
    <FormCheckbox
      checked={false}
      id="checkbox"
      onChange={[Function]}
    />
    <label
      htmlFor="checkbox"
      id="heart"
    >
      <img
        src="https://static5.depositphotos.com/1001759/463/i/600/depositphotos_4634746-stock-photo-red-heart-isolated-white-background.jpg"
        width="50px"
      />
    </label>
    <h5>
      Oppskrift:
    </h5>
    <pre>
      Bland deigen og la den heve
    </pre>
    <h3>
      Ingredienser
    </h3>
    Porsjoner
     
    <ButtonDanger
      id="btnDec"
      onClick={[Function]}
    >
      -
    </ButtonDanger>
     
    <b>
      5
    </b>
     
    <ButtonSuccess
      id="btnInc"
      onClick={[Function]}
    >
      +
    </ButtonSuccess>
    <p
      key="0"
    >
      1
      .
       
      pizzadeig
       
      1.3
       
      stk
    </p>
    <p
      key="1"
    >
      2
      .
       
      pizza fyll
       
      1.3
       
      håndfull
    </p>
    <p
      key="2"
    >
      3
      .
       
      kjøttboller
       
      500.0
       
      g
    </p>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Endre oppskrift
  </ButtonSuccess>
  <ButtonDanger
    id="deleteRecipe"
    onClick={[Function]}
  >
    Slett oppskrift
  </ButtonDanger>
  <ButtonSuccess
    id="btnSend"
    onClick={[Function]}
  >
    Send ingredienser til handleliste
  </ButtonSuccess>
</div>
`;

exports[`ShowRecipe tests Number of likes 1`] = `
<div>
  <Card
    title=""
  >
    <img
      src="pizza.jpg"
    />
    <h1>
      Pizza
    </h1>
    <p>
      Beskrivelse: 
      Pizza er god og enkel
    </p>
    <p>
      Kategori:
       
      enkelt
    </p>
    <p>
      Antall likes: 
      1
    </p>
    <FormCheckbox
      checked={false}
      id="checkbox"
      onChange={[Function]}
    />
    <label
      htmlFor="checkbox"
      id="heart"
    >
      <img
        src="https://static5.depositphotos.com/1001759/463/i/600/depositphotos_4634746-stock-photo-red-heart-isolated-white-background.jpg"
        width="50px"
      />
    </label>
    <h5>
      Oppskrift:
    </h5>
    <pre>
      Bland deigen og la den heve
    </pre>
    <h3>
      Ingredienser
    </h3>
    Porsjoner
     
    <ButtonDanger
      id="btnDec"
      onClick={[Function]}
    >
      -
    </ButtonDanger>
     
    <b>
      4
    </b>
     
    <ButtonSuccess
      id="btnInc"
      onClick={[Function]}
    >
      +
    </ButtonSuccess>
    <p
      key="0"
    >
      1
      .
       
      pizzadeig
       
      1.0
       
      stk
    </p>
    <p
      key="1"
    >
      2
      .
       
      pizza fyll
       
      1.0
       
      håndfull
    </p>
    <p
      key="2"
    >
      3
      .
       
      kjøttboller
       
      400.0
       
      g
    </p>
  </Card>
  <ButtonSuccess
    onClick={[Function]}
  >
    Endre oppskrift
  </ButtonSuccess>
  <ButtonDanger
    id="deleteRecipe"
    onClick={[Function]}
  >
    Slett oppskrift
  </ButtonDanger>
  <ButtonSuccess
    id="btnSend"
    onClick={[Function]}
  >
    Send ingredienser til handleliste
  </ButtonSuccess>
</div>
`;
